### ISSUE TITLE
create edge with same subquery between from and to use position params, result is wrong

### ISSUE THREAD

**Issue by robinZhao** [orientechnologies/orientdb#8809]
## Expected behavior  
```java
public class TestOrientClient {
    public static void main(String[] args){
        OrientDB orientDB = new OrientDB("remote:localhost/test", OrientDBConfig.defaultConfig());
        try(ODatabaseSession db = orientDB.open("test","root","root");) {
            // Enter your code here...
            db.begin(OTransaction.TXTYPE.OPTIMISTIC);
            db.command("create edge friends from (select from person where name=?) to (select from person where name=?)","tom","green");
            db.commit();
        }
    }
}
```

exist data in person class
```code
+----+-----+------+-----+----+----+
|#   |@RID |@CLASS|name |age |id  |
+----+-----+------+-----+----+----+
|0   |#45:0|person|tom  |16  |1   |
|1   |#46:0|person|green|16  |1   |
+----+-----+------+-----+----+----+
```
result  should be:
```code
+----+-----+-------+-----+-----+
|#   |@RID |@CLASS |out  |in   |
+----+-----+-------+-----+-----+
|0   |#49:0|friends|#45:0|#46:0|
+----+-----+-------+-----+-----+
```
## Actual behavior  
```code
+----+-----+-------+-----+-----+
|#   |@RID |@CLASS |out  |in   |
+----+-----+-------+-----+-----+
|0   |#49:0|friends|#45:0|#45:0|
+----+-----+-------+-----+-----+
```
## Steps to reproduce  
<add here>  
<An SQL script to reproduce the problem or a JUnit test case will increase **A LOT** the chance to have a quick fix>

```code

orientdb {db=test}> create class person extends V;
orientdb {db=test}> create class friends extends E;
orientdb {db=test}> insert into person (name,age,id) values ('tom',16,1);
orientdb {db=test}> insert into person (name,age,id) values ('green',16,1);
``` 
```java 
try(ODatabaseSession db = orientDB.open("test","root","root");) {
            // Enter your code here...
            db.begin(OTransaction.TXTYPE.OPTIMISTIC);
            db.command("create edge friends from (select from person where name=?) to (select from person where name=?)","tom","green");
            db.commit();
        }
```

**Comment by luigidellaquila** [comment]
Hi @robinZhao 

Thank you for reporting, it could be a problem with the query execution plan caching, I'm checking it now.
In the meantime, you can work-around it using named parameters

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @robinZhao 

I managed to reproduce and fix the problem, the fix will be released with v 3.0.18

Thanks

Luigi
